### ISSUE TITLE
Parsing error if the username contains slash

### ISSUE THREAD

**Issue by tyldum** [Nordaaker/convos#315]
ZNC uses "username/network" for usernames.
When this is parsed back the server address becomes the username.

**Comment by jhthorsen** [comment]
Can you give further details? I don't follow.

1. Where are you, which fields do you fill in?
2. Do you get an error message?
3. Any other details..?

**Comment by tyldum** [comment]
url parameter in connection.json
Further testing shows it's a Mojo::URL issue.

my $url = Mojo::URL->new('irc://username/network:password@example.com:3000); 
# $url->host is now "username"

**Comment by jhthorsen** [comment]
Ah! Thank you. Wanna make a PR? I'm not sure if I have time to look at this today.

I guess this line need to be changed: https://github.com/Nordaaker/convos/blob/master/assets/vue/convos-connection-settings.vue#L102

Into something like this.
```
var userinfo = [this.username, this.password].map(function(str) {
  return encodeURIComponent(str);
}).join(":");
```

**Comment by jhthorsen** [comment]
If you want to try it out, you might need to install the developer deps:

```
./script/convos install --develop
MOJO_ASSETPACK_LAZY=1 ./script/convos dev
```

`MOJO_ASSETPACK_LAZY` is used so you can make changes in to the files in `assets/` and see the changes in your browser without restarting the server.

Ask in #convos, if you need a test IRC server.

**Comment by tyldum** [comment]
As pointed out by @ksmadsen on IRC, the slash should be encoded. Which means the error is in Convos when parsing/storing user input.

The suggested solution did not work, but I ran out of time to debug further. Might do so later if still open.

**Comment by jhthorsen** [comment]
"suggested solution did not work": The solution I posted above?

**Comment by tyldum** [comment]
I need to look at when I have better time. When testing the websocket staeted acting up, even a git reset --hard has the same issue. I have to refresh the page after I add a server - it doesn't update.
Even running directly without the reverse proxy breaks, but I don't have time right now.

And now my VPN went down, so the box I worked on is not accessible. I'll have to revisit this later.

**Comment by jhthorsen** [comment]
The result was quite a big commit, but I hope it's fixed in aa0daa3. Please let me know if you can try out the "master" branch.

The reason for so many changes is simply that username/password was not tested and seems to have been completely broken to me, not just related to the `encodeURIComponent()` bug

**Comment by jhthorsen** [comment]
@tyldum have confirmed that this is fixed in https://github.com/Nordaaker/convos/commit/aa0daa345433b604899cb7f9affc7904820e505a
